src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -1073,7 +1073,14 @@ private static void ReportImplicitImplementationMatchDiagnostics(Symbol interfac
             if (implicitImpl.ContainsTupleNames() && MemberSignatureComparer.ConsideringTupleNamesCreatesDifference(implicitImpl, interfaceMember))
             {
                 // it is ok to implement implicitly with no tuple names, for compatibility with C# 6, but otherwise names should match
-                diagnostics.Add(ErrorCode.ERR_ImplBadTupleNames, implicitImpl.Locations[0], implicitImpl, interfaceMember);
+                if (implicitImpl.ContainingType == implementingType)
+                {
+                    diagnostics.Add(ErrorCode.ERR_ImplBadTupleNames, implicitImpl.Locations[0], implicitImpl, interfaceMember);
+                }
+                else
+                {
+                    diagnostics.Add(ErrorCode.ERR_ImplBadTupleNames, implementingType.Locations[0], implicitImpl, interfaceMember);
```

Comment:
>implementingType.Locations[0] [](start = 69, length = 29)

I think the location should be the interface reference in the base list, we probably already have a helper that finds it.

Reply:
Take a look at the logic around error location in ReportImplicitImplementationMismatchDiagnostics method in this file. See if we can share it with this method.